ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoProvider.java

```diff
@@ -240,8 +240,11 @@ private void mergeRemedialActionsResults(Crac crac, RaoResult preventiveRaoResul
         });
     }
 
-    private void deleteCurativeVariants(Crac crac, Map<State, RaoResult> curativeRaoResults) {
-        curativeRaoResults.values().stream().map(RaoResult::getPostOptimVariantId).forEach(variantId ->
-            crac.getExtension(ResultVariantManager.class).deleteVariant(variantId));
+    private void deleteCurativeVariants(Crac crac, String preventivePostOptimVariantId, Map<State, RaoResult> curativeRaoResults) {
+        curativeRaoResults.values().stream().map(RaoResult::getPostOptimVariantId).forEach(variantId -> {
+            if (!variantId.equals(preventivePostOptimVariantId)) {
```

Comment:
use filter instead ?
and can you add a unit test for this method ?

Reply:
done

Reply:
sorry not unit test